### PR TITLE
Replace django-sslify with SecurityMiddleware.

### PR DESCRIPTION
**Pull request by glogiotatidis**
- [ ] Update `SECURE_SSL_REDIRECT` and `DISABLE_SSL` env variables in Deis
- [ ] Update HSTS env variables in Deis
- [ ] Update `SECURE_BROWSER_XSS_FILTER` 
- [ ] Update `SECURE_CONTENT_TYPE_NOSNIFF`

### CODE REVIEW

File: careers/settings.py [hunk 1]
```diff
@@ -200,9 +200,14 @@
...
 DEAD_MANS_SNITCH_URL = config('DEAD_MANS_SNITCH', default=None)
 
-SSLIFY_DISABLE = config('DISABLE_SSL', default=DEBUG, cast=bool)
-SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
 USE_X_FORWARDED_HOST = True
+SECURE_SSL_REDIRECT = config('SECURE_SSL_REDIRECT', default=not DEBUG, cast=bool)
+SECURE_HSTS_SECONDS = config('SECURE_HSTS_SECONDS', default='0', cast=int)
+SECURE_HSTS_INCLUDE_SUBDOMAINS = False
+SECURE_BROWSER_XSS_FILTER = config('SECURE_BROWSER_XSS_FILTER', default=False, cast=bool)
+SECURE_CONTENT_TYPE_NOSNIFF = config('SECURE_CONTENT_TYPE_NOSNIFF', default=False, cast=bool)
```
Comment (glogiotatidis) [inline careers/settings.py]: @pmac I didn't see this in bedrock's PR. Maybe interesting?

**Comment by glogiotatidis** [comment]
@pmac r?

File: careers/settings.py (hunk 1 above)
Comment (pmac) [inline careers/settings.py]: I looked at it. It seems to mostly be useful if the server ever deals with user uploaded files, which bedrock doesn't. I guess it couldn't hurt though.

File: careers/settings.py [hunk 2]
```diff
@@ -200,9 +200,14 @@
...
+if SECURE_SSL_REDIRECT:
```
Comment (pmac) [inline careers/settings.py]: I think we do this in bedrock, but I'm not sure why. I don't think the proxy header setting has any affect if the requests isn't secure.

**Comment by pmac** [comment]
r+ 🙌

File: careers/settings.py (hunk 2 above)
Comment (glogiotatidis) [inline careers/settings.py]: Aha, I didn't have the check  before and it didn't cause any harm. I'll revert to what I had before.

**Pull request merged**